### PR TITLE
ENH: Add dynamic dropdown menu

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

Adds a dynamic dropdown in anticipation of templates of other species. Changes include:
* Addition of `validator/<species>` route to grab template names via JS
* Removes old `human_template` from route as it is no longer necessary.
* Adds `form_choices` for dropdown options, which is then used to dynamically populate 2nd dropdown menu with templates
  * 1st dropdown can be selected with `fid_species` - choices are automatically pulled by directory names under `afids-templates/`
  * 2nd dropdown reuses the old `fid_templates` - automatically pulled by templates stored in `afids-templates/<species>/`
* Moves validation to 2nd dropdown (will appear for each species) 

Some UI tweaks may be necessary and any new species will need to be linked with a db. Would recommend holding off on merging until necessary (templates from new species) and will likely include a rebase from pending PRs. Local tests indicate all other functionality is working as expected. 

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [x] Code has been run through [`black`](https://github.com/psf/black) with the `-l 79` flag.
- [x] I have included necessary documentation or comments (as necessary)
- [x] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

_PR template was adopted from [appium](https://github.com/appium/appium/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_

### CODE REVIEW

File: templates/validator.html [hunk 1]
```diff
@@ -15,23 +15,20 @@
...
+            <fieldset class="form-group">
+              <legend>Select a template to compare against.</legend>
+              <select id="species" class="form-control" name="fid_species" method="GET" action="/">
+                {% for choice in form_choices %}
+                <option value="{{choice}}">{{choice}}</option>
+                {% endfor %}
+              </select>
+              <select id="templates" class="form-control" name="fid_template" method="GET" action="/">
+                <!-- Will autoload on page ready, start with blank option -->
+                <option value=""></option>
```
Comment (kaitj) [inline templates/validator.html:29]: There is probably a way to have this `<option value=""></option> added via js, but haven't figured that out yet. This is a workaround.

File: controller.py [hunk 2]
```diff
@@ -227,6 +226,22 @@ def validator():
     )
 
 
+@app.route("/validator/<species>")
+def get_templates(species):
+    """Get templates corresponding to specific species"""
+    # Alawys include a validate object
+    template_obj = {}
+    template_obj["name"] = "Validate file structure"
+    templates = [template_obj]
```
Comment (tkkuehn) [inline controller.py:235]: I would just simplify these three lines to `templates = [{"name": "Validate file structure"}]`

File: controller.py [hunk 3]
```diff
@@ -227,6 +226,22 @@ def validator():
...
+    for species_templates in os.listdir(f"{AFIDS_DIR}/{species.lower()}"):
+        if "tpl" in species_templates:
+            template_obj = {}
+            template_obj["name"] = species_templates[4:].split("_")[0]
+            templates.append(template_obj)
```
Comment (tkkuehn) [inline controller.py:240]: Similarly, could just do `templates.append({"name": species_templates[4:].split("_")[0]})`

File: controller.py [hunk 4]
```diff
@@ -227,6 +226,22 @@ def validator():
...
+
+    return jsonify({"templates": templates})
```
Comment (tkkuehn) [inline controller.py:242]: Could we just make this a list of the template names, instead of a dictionary with a list of singleton dictionaries?

Comment (tkkuehn) [reply controller.py:242]: Would have to change the JS a bit in this case but it seems doable to me.

**Review by tkkuehn — COMMENTED** [review]
Basically looks good to me, just a couple of nitpicks with the new `get_templates` function.

File: controller.py (hunk 4 above)
Comment (tkkuehn) [reply controller.py:242]: So this whole function could be something to the effect of:

```
return ["Validate file structure"] + [
    species_templates[4:].split("_")[0]
    for species_templates in os.listdir(f"{AFIDS_DIR}/{species.lower()}")
    if "tpl" in species_templates
]
```

**Review by tkkuehn — APPROVED** [review]
Looks good, thanks!

**Pull request merged**